java/java-impl/src/com/intellij/codeInsight/FunctionalInterfaceSuggester.java

```diff
@@ -139,27 +139,29 @@
             PsiType substitutedThrowable = substitutor.substitute(thrownType);
             if (codeBlock == null || !(substitutedThrowable instanceof PsiClassType) ||
                 !ExceptionUtil.isHandled((PsiClassType)substitutedThrowable, codeBlock)) {
-              return null;
+              return Collections.emptyList();
             }
           }
         }
 
         final PsiElementFactory elementFactory = JavaPsiFacade.getElementFactory(aClass.getProject());
-        return elementFactory.createType(aClass, substitutor);
+        return Collections.singletonList(elementFactory.createType(aClass, substitutor));
       }
-      return null;
+      return Collections.emptyList();
     });
   }
 
-  private static <T extends PsiElement> Collection<? extends PsiType> suggestFunctionalInterfaces(final @NotNull T element, final NullableFunction<? super PsiClass, ? extends PsiType> acceptanceChecker) {
+  private static <T extends PsiElement> Collection<? extends PsiType> suggestFunctionalInterfaces(final @NotNull T element,
+                                                                                                  final Function<? super PsiClass, List<? extends PsiType>> acceptanceChecker) {
     final Project project = element.getProject();
     final Set<PsiType> types = new HashSet<>();
     final Processor<PsiMember> consumer = member -> {
-      if (member instanceof PsiClass && Java15APIUsageInspection.getLastIncompatibleLanguageLevel(member, PsiUtil.getLanguageLevel(element)) == null) {
+      if (member instanceof PsiClass &&
+          Java15APIUsageInspection.getLastIncompatibleLanguageLevel(member, PsiUtil.getLanguageLevel(element)) == null) {
         if (!JavaPsiFacade.getInstance(project).getResolveHelper().isAccessible(member, element, null)) {
           return true;
         }
-        ContainerUtil.addIfNotNull(types, acceptanceChecker.fun((PsiClass)member));
+        ContainerUtil.addAll(types, acceptanceChecker.apply((PsiClass)member));
```

Comment:
I have changes suggester signature to return list of types, because I am suggesting every suitable ctor when expression is constructor method reference.